doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -422,14 +422,153 @@ int main(void)
 </div>
 
 <div id="app-java" class="tab-pane fade" markdown="1" >
+```java
+
+import org.kaaproject.kaa.client.DesktopKaaPlatformContext;
+import org.kaaproject.kaa.client.Kaa;
+import org.kaaproject.kaa.client.KaaClient;
+import org.kaaproject.kaa.client.SimpleKaaClientStateListener;
+import org.kaaproject.kaa.client.configuration.base.ConfigurationListener;
+import org.kaaproject.kaa.client.configuration.base.SimpleConfigurationStorage;
+import org.kaaproject.kaa.client.logging.strategies.RecordCountLogUploadStrategy;
+import org.kaaproject.kaa.schema.sample.Configuration;
+import org.kaaproject.kaa.schema.sample.DataCollection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Class implement functionality for First Kaa application
+ *
+ * @author Maksym Liashenko
+ */
+public class FirstDemo {
+
+    private static final long DEFAULT_SAMPLE_TIME = 1000L;
+
+    private static final Logger LOG = LoggerFactory.getLogger(FirstDemo.class);
+    private static Timer timer = new Timer();
+    private static KaaClient kaaClient;
+
+
+    public static void main(String[] args) {
+        LOG.info("Kaa First Demo app starting!");
+        startKaaClient();
+
+        LOG.info("--= Press any key to exit =--");
+        try {
+            System.in.read();
+        } catch (IOException e) {
+            LOG.error("IOException has occurred: " + e.getMessage());
+        }
+        LOG.info("Stopping...");
 
-</div>
+        timer.cancel();
+        stopKaaClient();
+    }
 
-<div id="app-obj-c" class="tab-pane fade" markdown="1" >
+    private static void startKaaClient() {
+        /*
+         * Create the Kaa desktop context for the application.
+         */
+        DesktopKaaPlatformContext desktopKaaPlatformContext = new DesktopKaaPlatformContext();
+
+        /*
+         * Create a Kaa client and add a listener which displays the Kaa client
+         * configuration as soon as the Kaa client is started.
+         */
+        kaaClient = Kaa.newClient(desktopKaaPlatformContext, new SimpleKaaClientStateListener() {
+            @Override
+            public void onStarted() {
+                super.onStarted();
+                LOG.info("Kaa client started");
+
+                Configuration configuration = kaaClient.getConfiguration();
+                LOG.info("Default sample period: " + configuration.getSamplePeriod());
```

Comment:
This is antipattern. Correct usage:  
`LOG.info("Default sample period: {}", configuration.getSamplePeriod());`